src/ui/BlockEditor.tsx

```diff
@@ -400,32 +391,28 @@ class BlockEditor extends Component<BlockEditorProps> {
           }
           if (annt === "") annt = "change";
           getState().undoableAction = annt; //?
-          commitChanges(changes, SHARED.parse, false, -1, this.newAST);
+          commitChanges(changes, SHARED.parse, false, -1, newAST);
         }
       }
     });
   };
 
-  /**
-   * @internal
-   * When the editor mounts, (1) set change event handlers and AST,
-   * (2) set the focus, (3) set aria attributes, and (4) build the API
-   */
-  handleEditorDidMount = (ed: Editor) => {
-    const { passedAST: ast, setAST, search, options, onMount } = this.props;
-    ed.on("beforeChange", this.handleBeforeChange);
-    ed.on("changes", this.handleChanges);
+  // When the editor mounts, (1) set change event handlers and AST,
+  // (2) set the focus, (3) set aria attributes, and (4) build the API
+  const handleEditorDidMount = (ed: Editor) => {
+    ed.on("beforeChange", handleBeforeChange);
+    ed.on("changes", handleChanges);
 
-    // set AST and searchg properties and collapse preferences
-    setAST(ast);
+    // set AST, search properties, and collapse preferences
+    dispatch({ type: "SET_AST", ast });
```

Comment:
This also looks like a problem. Before `ast` was `this.props.passedAST`, now it is the ast from the root state.

Reply:
Aha! YES! *thank you*